### DISCUSSION TITLE
fastroute: What is the minimum number of metal layers OpenROAD can route in?

### DISCUSSION THREAD

**Discussion by Sleighbells64** [The-OpenROAD-Project/OpenROAD#4165] [Q&A]
My apologies if this is an obvious question. I am working with a fab trying to develop a new process that currently has 2 metal layers, and we were wondering if we could use OpenROAD/OpenLANE to build test chips. I was unable to find an answer in the documentation, but if it is there I would love to know where to look. Thank you for your help!

**Comment by maliberty** [comment]
In theory we could route in two layers but we've never tried it.  We don't have a specialized channel router.  My expectation is that we will hit some issues and have to iterate.  If the PDK is closed it will be harder to do so.

**Comment by JackFz** [comment]
I'm sure that Fastroute and TritonRoute are capable of routing on two layers.

**Comment by steux** [comment]
Yes, I confirm that FastRoute and TritonRoute are capable of routing on two layers. However, as it is, global router will crash as it takes layer 3 as a reference for estimating guides. Here is the patch I had to apply to make it work with XFab XH035 MET2 porocess :
```
diff --git a/src/grt/src/GlobalRouter.cpp b/src/grt/src/GlobalRouter.cpp
index b2eade75e..6fea47a60 100644
--- a/src/grt/src/GlobalRouter.cpp
+++ b/src/grt/src/GlobalRouter.cpp
@@ -93,7 +93,7 @@ GlobalRouter::GlobalRouter()
       groute_renderer_(nullptr),
       grid_(new Grid),
       adjustment_(0.0),
-      layer_for_guide_dimension_(3),
+      layer_for_guide_dimension_(2), // SandCastle : Ho!!! 3 is not compatible with XH035
       congestion_report_iter_step_(0),
       allow_congestion_(false),
       macro_extension_(0),
@@ -2004,6 +2004,7 @@ void GlobalRouter::initGridAndNets()
   if (routing_layers_.empty()) {
     int min_layer, max_layer;
     getMinMaxLayer(min_layer, max_layer);
+    fprintf(stderr, "Min & Max routing layer: %d, %d\n", min_layer, max_layer);

     initRoutingLayers(min_layer, max_layer);
     initRoutingTracks(max_layer);
```

**Reply by maliberty** [reply]
@eder-matheus I think taking the min of 3 and the max routing layer for layer_for_guide_dimension_ would accommodate two layer processes based on this comment.
